### PR TITLE
feat: Enhance driver robustness, data types, and testing

### PR DESCRIPTION
**Pull request by coji**
This pull request introduces a wide range of improvements and bug fixes across the driver, introspector, and helper utilities to enhance robustness, correctness, and the developer experience.

**Key Changes:**

*   **Driver (`driver-wasm.ts`)**
    *   **Reliable Result Parsing**: The logic for parsing query results has been completely overhauled. It now correctly distinguishes between DML statements (returning an affected row count) and `SELECT` or `RETURNING` queries (returning rows), fixing issues where DML counts were not reported.
    *   **Improved Data Type Conversion**: Type conversion from Arrow to JS is now significantly more robust:
        *   **`BIT` vs `BLOB`**: The driver now uses Arrow metadata to reliably distinguish between `BIT` and `BLOB` types, falling back to heuristics only when necessary.
        *   **`TIMESTAMP`/`DATE`**: Parsing is more flexible, correctly handling various string formats, UTC conversion, and truncating microsecond precision to milliseconds.
        *   **`BLOB`**: The `blob` helper now accepts `Uint8Array`, `ArrayBufferLike`, or `number[]`.
    *   **Resource Management**: Fixed a critical bug where prepared statements were not being closed, which could lead to resource leaks. Statements are now reliably closed in `finally` blocks.
    *   **Connection Management**: `init()` and `destroy()` are now idempotent and safer to call multiple times.

*   **Introspector (`introspector.ts`)**
    *   Fixed a major bug in table metadata parsing that caused columns not to be grouped correctly by table and could lead to errors when mutating frozen objects. The introspector now reliably returns accurate table structures.

*   **Query Compiler (`query-compiler.ts`)**
    *   Fixed a bug where a table alias (e.g., `selectFrom('my_table as t')`) would be dropped if `tableMappings` was used for that table. Aliases are now correctly preserved.

*   **Helpers & Extensions**
    *   **`createTablesAsSelect`**: Now validates table names to prevent invalid identifiers and potential SQL injection.
    *   **`struct` helper**: Now accepts plain JS values (in addition to `sql` builders) and validates that struct keys are valid identifiers.

*   **Comprehensive Testing**
    *   Added new test suites for the introspector, query compiler, and `createTablesAsSelect` extension.
    *   Significantly expanded existing tests for `SELECT`, `INSERT`, `UPDATE`, and data type round-tripping to cover the new fixes and edge cases.
    *   Test setup now correctly uses DuckDB's virtual file system to load test data, aligning with browser/WASM best practices.

*   **Housekeeping**
    *   Renamed `DuckDBNodeDataTypes` to `DuckDBWasmDataTypes` to better reflect the package's focus.
    *   Updated `README.md` and JSDoc examples to be more accurate and reflect current usage.

## Summary by CodeRabbit

* **New Features**
  * Improved result handling for BIT/BLOB/MAP/STRUCT/LIST and safer CTAS (identifier validation).

* **Bug Fixes**
  * Reliable statement/resource cleanup; accurate affected-row counts; corrected timestamp/timezone and blob/bit handling; introspector queries aligned with DuckDB.

* **Documentation**
  * README examples updated for WASM setup, APIs, and consistent formatting.

* **Refactor**
  * Public datatype alias renamed to DuckDBWasmDataTypes.

* **Tests**
  * Added/expanded tests for returning rows, delete counts, blobs, types, compiler aliases, and introspection.

### CODE REVIEW

File: src/helper/datatypes.ts [hunk 1]
```diff
@@ -1,6 +1,6 @@
 import { type RawBuilder, sql } from "kysely";
```
[the remark this answers is not in the document]
Comment (coji) [reply src/helper/datatypes.ts:2]: いらん
